src/main/java/com/launchdarkly/client/FeatureFlag.java

```diff
@@ -51,9 +53,6 @@ static FeatureFlag fromJson(String json) {
   }
 
   EvalResult evaluate(LDUser user, FeatureStore featureStore) throws EvaluationException {
-    if (user == null || user.getKey() == null) {
```

Comment:
I pushed this up into the client code so it happens first thing. This is more consistent with the go client behavior and avoids a call to the feature store.